cinema/views.py

```diff
@@ -97,12 +99,27 @@ def get_queryset(self):
     def get_serializer_class(self):
         if self.action == "list":
             return MovieListSerializer
-
         if self.action == "retrieve":
             return MovieDetailSerializer
+        if self.action == "upload_image":
+            return MovieImagesSerializer
 
         return MovieSerializer
 
+    @action(
+        methods=["POST"],
+        detail=True,
+        permission_classes=[IsAdminUser],
+        url_path="movie-upload-image",
+    )
+    def upload_image(self, request, pk=None):
+        movie = self.get_object()
+        serializer = self.get_serializer(movie, data=request.data)
+        if serializer.is_valid(raise_exception=True):
```

Comment:
you don't need `if` and response 400, if you use serializer.is_valid(raise_exception=True)

Reply:
try to avoid it

Reply:
fixied,thanks!